modules/features2d/src/kaze/AKAZEFeatures.cpp

```diff
@@ -342,14 +342,14 @@ void AKAZEFeatures::Find_Scale_Space_Extrema(std::vector<KeyPoint>& kpts)
 
             if (is_out == false) {
               if (is_repeated == false) {
-                point.pt.x *= ratio;
-                point.pt.y *= ratio;
+                point.pt.x = point.pt.x*ratio + .5*(ratio-1.0);
+                point.pt.y = point.pt.y*ratio + .5*(ratio-1.0);
```

Comment:
`point.pt.x` and `point.pt.y` are both explicitly casted to float and you are pushing in a double value due to the mulitplying of .5. Solution to this problem can be to add an explicit cast.

point.pt.x = (float)(point.pt.x*ratio + .5*(ratio-1.0));
point.pt.y = (float)(point.pt.y*ratio + .5*(ratio-1.0));

But the question still remains if we will ever run into precision loss due to this change.